extensions/blocks/dialogue/dialogue.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Dialogue Block.
+ *
+ * @since 9.3.0
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Dialogue;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Gutenberg;
+
+const FEATURE_NAME = 'dialogue';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	$deprecated = function_exists( 'gutenberg_get_post_from_context' );
+	$uses       = $deprecated ? 'context' : 'uses_context';
+	Blocks::jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'render_callback' => __NAMESPACE__ . '\render_block',
+			$uses             => array(
+				'jetpack/conversation-participants',
+				'jetpack/conversation-showtimestamp',
+			),
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Dialogue block registration/dependency declaration.
+ *
+ * @param array  $attrs    Array containing the Dialogue block attributes.
+ * @param string $block_content String containing the Dialogue block content.
+ * @param object $block Block object data.
+ *
+ * @return string
+ */
+function render_block( $attrs, $block_content, $block ) {
+	Jetpack_Gutenberg::load_assets_as_required( FEATURE_NAME );
+
+	$default_participants = json_decode(
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		file_get_contents( JETPACK__PLUGIN_DIR . 'extensions/blocks/conversation/participants.json' ),
+		true
+	);
+
+	// Attributes..
+	$participant_slug_attr  = isset( $attrs['participantSlug'] ) ? $attrs['participantSlug'] : null;
+	$participant_label_attr = isset( $attrs['participant'] ) ? $attrs['participant'] : null;
+	$timestamp              = isset( $attrs['timeStamp'] ) ? esc_attr( $attrs['timeStamp'] ) : '00:00';
+	$is_custom_spaker       = $participant_label_attr && ! $participant_slug_attr;
```

Comment:
I think you meant `speaker` here and below?

```suggestion
	$is_custom_speaker       = $participant_label_attr && ! $participant_slug_attr;
```